core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -1644,13 +1677,52 @@ class KafkaZkClient private (zooKeeperClient: ZooKeeperClient, isSecure: Boolean
       }
     }
 
+    // This method is part of the work around done in the KAFKA-7165, once ZOOKEEPER-2985 is complete, this code must
+    // be deleted, if by the time you are reading this the code is still present and the bug solve, please delete it.
+    private def delete(): Code = {
+      val deleteRequest = DeleteRequest(path, ZkVersion.MatchAnyVersion)
+      val deleteResponse = retryRequestUntilConnected(deleteRequest)
+      deleteResponse.resultCode match {
+        case code@ Code.OK => code
+        case code =>
```

Comment:
It seems that if code is NoNode, we should just let it go.

Reply:
@junrao sorry, but I do not get this comment.
Is true (after you point this out) that at the moment we try to delete the ephemeral znode, it could not be present anymore an a `NoNode` response will be received, and because we are just checking `if (codeAfterDelete == Code.OK)` the `created()` call will not be performed, hence, the Broker will be not registered. Is that what you meant?

If so, will be adding `if (codeAfterDelete == Code.OK || codeAfterDelete == Code.NONODE)`?

Hope you can help me to understand your point.

Cheers!

Reply:
Or this?
```
deleteResponse.resultCode match {
        case code@ Code.OK => code
        

      > case code@ Code.NONODE => code

        case code =>
          error(s"Error while deleting ephemeral node at $path with return code: $code")
          code
}
```

Reply:
Yes, it seems that if the node doesn't exist (ZK server could have deleted), we should still do the create() instead of returning an error code.